### PR TITLE
Ticket #1031: Update Staff permissions for contacts, websites, addresses, domain information and domain application

### LINKED ISSUE #1031: Django admin: Enable editing of Contacts, Websites, & Addresses (Staff Users)
### Issue Description

Currently, Staff users cannot edit Contacts or Websites in Django admin. We need to allow editing but limit some of the functionality as noted in the acceptance criteria.

As part of this ticket, we should also update a few fields on the Domain Application / Information pages that should be editable for Staff.

Staff = Analysts

**Acceptance Criteria**

- [x] Allow Staff to edit Websites 
- [x] Allow Staff to edit Contacts
- [x] Do not allow Staff the ability to "delete" Contacts
- [x] Disable the "Users" field on the Contacts page for Staff (make it view-only)
- [x] Allow Staff to edit Organization Address field 1, Address field 2 and Zip Code on the Domain Application page
- [x] Allow Staff to edit Organization Address field 1, Address field 2 and Zip Code on the Domain Information page

### Additional Context (optional)

This is just permissions for Staff users.

### Issue Links

Should be done after #1027, as it will facilitate this functionality

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #1031

## Changes
(I've added in here + Setup check boxes for peace of mind cause I know there's multiple changes)

Staff == Analyst 
- [ ] Allow Staff to edit Websites
- [ ] Allow Staff to edit Contacts
- [ ]  Do not allow Staff the ability to "delete" Contacts
- [ ] Disable the "Users" field on the Contacts page for Staff (make it view-only)
- [ ] Allow Staff to edit Organization Address field 1, Address field 2 and Zip Code on the Domain Application page
- [ ] Allow Staff to edit Organization Address field 1, Address field 2 and Zip Code on the Domain Information page

## Setup

1. Sign into https://getgov-rh.app.cloud.gov/admin/ as an analyst
2. Log into admin page
3. As a analyst you should now:
- [x] Be able to see the Websites column
- [x]  For the Contacts page,  you should be able to edit the Contacts now but NOT able to delete Contacts (you should NOT see a red button on the bottom right)
- [x] For the Contacts page, you should NOT see a dropdown in the "Users" field
- [x]  For the Domain Application page, should be able to edit Organization Address field 1, Address field 2 and Zip Code
- [x]  For the Domain Information page, should be able to edit Organization Address field 1, Address field 2 and Zip Code

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [x] Checked that all code is adequately covered by tests
- [x] Made it clear which comments need to be addressed before this work is merged
- [x] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/admin.py [hunk 1]
```diff
@@ -294,6 +294,26 @@ def contact(self, obj: models.Contact):
...
+        "user",
+    ]
+
+    def get_readonly_fields(self, request, obj=None):
+        """Set the read-only state on form elements.
+        We have 1 conditions that determine which fields are read-only:
+        admin user permissions.
+        """
+
+        readonly_fields = list(self.readonly_fields)
```
Comment (zandercymatics) [inline src/registrar/admin.py:308]: (Q) why are we doing a list conversion here? Is this of type dict?

**Review by zandercymatics — COMMENTED** [review]
Might be a product question, but do we want to keep this password field for analysts? I can't get it to populate with data in either event.
image

**Review by zandercymatics — APPROVED** [review]
Looks great! Your changes work perfectly, couldn't break them when I tried. 

The main thing that this is missing is some test cases on if the right fields display. I think it depends on the complexity, so if its a lot of development time + effort to implement we can split that off into another ticket, but if the changes are not that large I would recommend including those in here.

Not blocking for that reason

**Review by erinysong — APPROVED** [review]
LGTM outside of Zander's question!

File: src/registrar/admin.py (hunk 1 above)
Comment (therealslimhsiehdy) [reply src/registrar/admin.py:308]: So I mirrored this from other sections, but I think you're right -- it's a type of dict and we want things in list form for the admin page!

**Comment by therealslimhsiehdy** [comment]
@zandercymatics + @erinysong Just updated this PR with tests, would love if you both have a time to re-review!

**Review by Katherine-Osos — APPROVED** [review]

**Pull request merged**